docs/compatibility.html

```diff
@@ -0,0 +1,206 @@
+ limitations under the License.
+-->
+
+<script>
+    <!--#include virtual="js/templateData.js" -->
+</script>
+
+<script id="compatibility-template" type="text/x-handlebars-template"></script>
+
+<div class="p-compatibility">
+    With the release of Kafka 4.0, significant changes have been introduced that impact compatibility across various
+    components. To assist users in planning upgrades and ensuring seamless interoperability, a comprehensive compatibility
+    matrix has been prepared.
+    <h4>JDK Compatibility Across Kafka Versions</h4>
+    <table>
+        <tr>
+            <th>Module</th>
+            <th>Kafka Version</th>
+            <th>Java 8</th>
+            <th>Java 11</th>
+            <th>Java 17</th>
+            <th>Java 23</th>
+        </tr>
+        <tr>
+            <td>Client</td>
+            <td>4.0.0</td>
+            <td>❌</td>
+            <td>✅</td>
+            <td>✅</td>
+            <td>✅</td>
+        </tr>
+        <tr>
+            <td>Streams</td>
+            <td>4.0.0</td>
+            <td>❌</td>
+            <td>✅</td>
+            <td>✅</td>
+            <td>✅</td>
+        </tr>
+        <tr>
+            <td>Connect</td>
+            <td>4.0.0</td>
+            <td>❌</td>
+            <td>❌</td>
+            <td>✅</td>
+            <td>✅</td>
+        </tr>
+        <tr>
+            <td>Server</td>
+            <td>4.0.0</td>
+            <td>❌</td>
+            <td>❌</td>
+            <td>✅</td>
+            <td>✅</td>
+        </tr>
+    </table>
+
+    <h4>Server Compatibility</h4>
+    <table>
+        <tr>
+            <th>KRaft Cluster Version</th>
+            <th>Compatibility 4.0 Server (dynamic voter)</th>
+            <th>Compatibility 4.0 Server (static voter)</th>
+        </tr>
+        <tr>
+            <td>before 3.2.x</td>
+            <td>❌</td>
+            <td>❌</td>
+        </tr>
+        <tr>
+            <td>3.3.x</td>
+            <td>❌</td>
+            <td>✅</td>
+        </tr>
+        <tr>
+            <td>3.4.x</td>
+            <td>❌</td>
+            <td>✅</td>
+        </tr>
+        <tr>
+            <td>3.5.x</td>
+            <td>❌</td>
+            <td>✅</td>
+        </tr>
+        <tr>
+            <td>3.6.x</td>
+            <td>❌</td>
+            <td>✅</td>
+        </tr>
+        <tr>
+            <td>3.7.x</td>
+            <td>❌</td>
+            <td>✅</td>
+        </tr>
+        <tr>
+            <td>3.8.x</td>
+            <td>❌</td>
+            <td>✅</td>
+        </tr>
+        <tr>
+            <td>3.9.x</td>
+            <td>✅</td>
+            <td>✅</td>
+        </tr>
+        <tr>
+            <td>4.0.x</td>
+            <td>✅</td>
+            <td>✅</td>
+        </tr>
+    </table>
+    <p>Note: Can’t upgrade server from static voter to dynamic voter, see <a href="https://issues.apache.org/jira/browse/KAFKA-16538">KAFKA-16538</a>.</p>
+
+    <h4>Client Compatibility</h4>
```

Comment:
```suggestion
    <h4>Client/Broker Forward Compatibility</h4>
```

Reply:
I believe "forward" is the right term here, as we are covering older client version and 4.0 brokers?